library/core/src/iter/traits/marker.rs

```diff
@@ -54,3 +54,8 @@ unsafe impl<I: TrustedLen + ?Sized> TrustedLen for &mut I {}
 /// [`try_fold()`]: Iterator::try_fold
 #[unstable(issue = "none", feature = "inplace_iteration")]
 pub unsafe trait InPlaceIterable: Iterator {}
+
+/// FIXME(jhpratt) docs
+#[unstable(feature = "trusted_step", issue = "none")]
+#[rustc_specialization_trait]
+pub unsafe trait TrustedStep: crate::iter::Step {}
```

Comment:
I was recently informed that unstable traits should either be non-public, `doc(hidden)` (if it's also needed in `alloc` or `std`) or have a tracking issue.

And is this really the right place? It's not a general marker trait for iterators. Well, I guess that's part of the ancient problem that Ranges shouldn't be iterators but are.

Is this trait even needed considering that `TrustedLen` is implemented for the same types? I.e. couldn't you just use `TrustedLen` as bound on the specialization?

Reply:
Wasn't sure on the location, I just threw it in with the rest of the marker traits. It's obviously trivial to move.

Why should it be non-public or hidden? Isn't the purpose to let people try it out on nightly?

Reply:
Depends on whether that is the intent. If it's only meant for std-internal use it needs to be hidden. If it's intended for external users then it should have a path to stabilization and thus needs a tracking issue.

Reply:
It's definitely intended for external usage, otherwise the change wouldn't have been requested. A tracking issue will certainly be created, but I don't think it's standard to do that before the feature is even merged?